lib/runners/processor/detekt.rb

```diff
@@ -0,0 +1,381 @@
+      capture3("mvn", maven_config[:goal], *mvn_options)
+      construct_result(
+        maven_config[:report_id],
+        maven_config[:report_path]
+      )
+    end
+
+    def construct_result(report_id, report_path)
+      output_file_path = current_dir / report_path
+      if output_file_path.exist?
+        trace_writer.message "Reading output from #{report_path}..."
+        output = output_file_path.read
+        trace_writer.message output
+      else
+        msg = "#{report_path} does not exist. Unexpected error occurred, processing cannot continue."
+        trace_writer.error msg
+        raise msg
+      end
+
+      switch_constructor(report_id, output)
+    end
+
+    def switch_constructor(report_id, output)
+      case report_id
+      when "xml"
+        construct_checkstyle_result(output)
+      when "txt"
+        construct_plain_result(output)
+      end
+    end
+
+    def construct_checkstyle_result(output)
+      document = REXML::Document.new(output)
+      unless document.root
+        msg = "Invalid XML output!"
+        trace_writer.error msg
+        raise msg
+      end
+
+      issues = []
+
+      document.root.each_element("file") do |file|
+        file.each_element do |error|
+          case error.name
+          when "error"
+            regexp = /detekt\.(.*)/
+            rule, = error[:source].match(regexp).captures
+
+            issues << construct_issue(
+              file: file[:name],
+              line: error[:line],
+              message: error[:message],
+              rule: rule,
+              severity: error[:severity]
+            )
+          else
+            add_warning error.text.strip, file: file[:name]
+          end
+        end
+      end
+
+      issues
+    end
+
+    def construct_plain_result(output)
+      issues = []
+
+      # Case of using txt reporter, they has 2 pattern like this:
+      # a) `NestedBlockDepth - 4/4 - [complex] at /path/to/ComplexClass.kt:9:13 - message...`
+      # b) `EmptyClassBlock - [FilteredClass] at /path/to/FilteredClass.kt:2:21 - message...`
+      regexp_a = /(.*) - .* - .* at (.*):(.*):(.*) - (.*)/
+      regexp_b = /(.*) - .* at (.*):(.*):(.*) - (.*)/
+      
+      output.lines(chomp: true).map do |line|
+        match = line.match(regexp_a)
+        if match.nil?
+          match = line.match(regexp_b)
+        end
+
+        unless match.nil?
+          rule, file, start_line, _, message = match.captures
+          issues << construct_issue(
+            file: file,
+            line: start_line,
+            message: message,
+            rule: rule
+          )
+        end
+      end
+
+      issues
+    end
+
+    def construct_issue(file:, line:, message:, rule:, severity: nil)
+      unless severity.nil?
+        object = { severity: severity }
+      end
+
+      Issue.new(
+        path: relative_path(file),
+        location: Location.new(start_line: line),
+        id: rule,
+        message: message,
+        object: object
```

Comment:
The following looks good. `unless severity.nil?` looks not needed.

```suggestion
        object: {
          severity: severity
        },
```